UI/obs-app.cpp

```diff
@@ -3294,6 +3294,8 @@ int main(int argc, char *argv[])
 				"--unfiltered_log: Make log unfiltered.\n\n"
 				"--disable-updater: Disable built-in updater (Windows/Mac only)\n\n"
 				"--disable-missing-files-check: Disable the missing files dialog which can appear on startup.\n\n"
+				"--debug-webrtc-whip: Enable webrtc WHIP logging\n"
+				"--debug-webrtc-filter <string>: Apply log filter string, default is INFO\n\n"
```

Comment:
This sounds like a common "log-level" parameter - a plugin should not introduce its own global logging parameters, instead it should adhere to the logging settings of the app itself.

Reply:
This is tricky because:
* OBS only has 4 levels of logging, debug and trace are often very different levels of verbosity
* Trace is too much, and requires filtering which OBS (afaik) doesn't offer
* OBS logs don't emit the concept of modules with which to filter by (e.g. I want to filter only webrtc_dtls::handshake code)
* Dependencies often have important information that may need to be surfaced selectively out of debug/trace

Is the objection to --debug-webrtc-whip? Which was a request to be implemented by tt?
Or to the filter? Which doesn't add _new_ log levels, merely allows you to filter log messages in rust
Or just that you can enable logs beyond the log-level?

WebRTC is tricky because it can fail in many hard to determine ways (network topology, ISP, ipv6, connectivity issues, etc.) and to log all of this at the info level would be noisy beyond usefulness. Thus the need to bring stuff out of the debug/trace range selectively is important.  Happy for feedback on this, I'm just not sure how this could work with the granularity of log levels offered and the inability to selectively pull important bits out of trace while debugging

Reply:
Additionally, much of the possible logging required to debug WebRTC issues contains arguably sensitive information that should not be wrapped into any normal log level settings. Uses should explicitly opt-in to these log messages in order to get them.

Reply:
We do not make any such distinction for other formats right now (which also expose sensitive information to the log). My main objection is to add specific flags for a specific plugin - the plugin needs to fit into OBS' architecture, not the other way around.

The purpose is sound, but then it should be implemented on an architecture level (i.e. add the trace level to OBS itself, then use it) instead of piling on features for a single plugin only.

Reply:
Fixed in 055879fc8e89fc16b7498f9e64e23295a74d1f86